### PR TITLE
Added get_config to custom layer in vision example

### PR DESCRIPTION
**Pull request by johko**
The example for Semi-Supervised Contrastive Learning with SimCLR is missing a `get_config` in the custom _RandomColorAffine_ class. 
This leads to:
`WARNING:tensorflow:Model failed to serialize as JSON. Ignoring... 
Layer RandomColorAffine has arguments ['self', 'brightness', 'jitter']
in `__init__` and therefore must override `get_config()`.` 

So I added the `get_config` method for the custom layer class

### CODE REVIEW

File: examples/vision/semisupervised_simclr.py [hunk 1]
```diff
@@ -173,6 +174,15 @@ def __init__(self, brightness=0, jitter=0, **kwargs):
 
         self.brightness = brightness
         self.jitter = jitter
+       
+    def get_config(self):
+
+        config = super().get_config().copy()
```
Comment (fchollet) [inline examples/vision/semisupervised_simclr.py:180]: No need for `copy()`

Also remove the blank line above

File: examples/vision/semisupervised_simclr.py [hunk 2]
```diff
@@ -197,14 +207,14 @@ def call(self, images, training=True):
...
-    zoom_factor = 1.0 - tf.sqrt(min_area)
+    zoom_factor = 1.0 - np.sqrt(min_area)
     return keras.Sequential(
         [
             keras.Input(shape=(image_size, image_size, image_channels)),
             layers.Rescaling(1 / 255),
             layers.RandomFlip("horizontal"),
             layers.RandomTranslation(zoom_factor / 2, zoom_factor / 2),
-            layers.RandomZoom((-zoom_factor, 0.0), (-zoom_factor, 0.0)),
+            # layers.RandomZoom((-zoom_factor, 0.0), (-zoom_factor, 0.0)),
```
Comment (fchollet) [inline examples/vision/semisupervised_simclr.py:217]: Uncomment

File: examples/vision/semisupervised_simclr.py [hunk 3]
```diff
@@ -197,14 +207,14 @@ def call(self, images, training=True):
 
 # Image augmentation module
 def get_augmenter(min_area, brightness, jitter):
-    zoom_factor = 1.0 - tf.sqrt(min_area)
+    zoom_factor = 1.0 - np.sqrt(min_area)
```
Comment (fchollet) [inline examples/vision/semisupervised_simclr.py:210]: It really is `tf`

File: examples/vision/semisupervised_simclr.py [hunk 4]
```diff
@@ -283,8 +293,10 @@ def get_encoder():
 )
 
 baseline_history = baseline_model.fit(
-    labeled_train_dataset, epochs=num_epochs, validation_data=test_dataset
+    labeled_train_dataset, epochs=1, validation_data=test_dataset
 )
+
+baseline_model.save(".")
```
Comment (fchollet) [inline examples/vision/semisupervised_simclr.py:299]: Revert these changes

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: examples/vision/semisupervised_simclr.py (hunk 3 above)
Comment (johko) [reply examples/vision/semisupervised_simclr.py:210]: I forgot to mention the reason I changed that. When using _tf.sqrt()_ and trying to save the model, there is a JSON serialization error:
`Unable to serialize 0.0669873058795929 to JSON. Unrecognized type <class 'tensorflow.python.framework.ops.EagerTensor'>.` 

Using _np.sqrt()_ instead was my workaround. I suppose there might be a better solution for this but my tried approaches did not solve the issue.

File: examples/vision/semisupervised_simclr.py (hunk 2 above)
Comment (johko) [reply examples/vision/semisupervised_simclr.py]: Yes, thank you. Forgot to push the clean up at first :sweat_smile:

File: examples/vision/semisupervised_simclr.py (hunk 3 above)
Comment (fchollet) [reply examples/vision/semisupervised_simclr.py:209]: Got it -- in that case consider using `math.sqrt` from the Python stdlib.

File: examples/vision/semisupervised_simclr.py [hunk 5]
```diff
@@ -283,8 +292,9 @@ def get_encoder():
 )
 
 baseline_history = baseline_model.fit(
-    labeled_train_dataset, epochs=num_epochs, validation_data=test_dataset
+    labeled_train_dataset, epochs=1, validation_data=test_dataset
```
Comment (fchollet) [inline examples/vision/semisupervised_simclr.py:295]: Revert this

**Review by fchollet — APPROVED** [review]
Thanks for the update! Please add the generated files.

File: .gitignore [hunk 6]
```diff
@@ -14,3 +14,5 @@ templates/examples/timeseries/*
 templates/examples/graph/*
 templates/guides/**/*.md
 templates/keras-tuner/getting_started.md
+wandb/*
+variables/*
```
Comment (johko) [inline .gitignore]: removed these custom entries in new commit

File: examples/vision/md/semisupervised_simclr.md [hunk 7]
```diff
@@ -1,878 +1,883 @@
...
-([Keras example](https://keras.io/examples/vision/simsiam/)):
-no momentum-encoder + no negatives
-
-In my experience, these methods are more brittle (they can collapse to a constant
-representation, I could not get them to work using this encoder architecture).
-Even though they are generally more dependent on the
-[model](https://generallyintelligent.ai/understanding-self-supervised-contrastive-learning.html)
-[architecture](https://arxiv.org/abs/2010.10241), they can improve
-performance at smaller batch sizes.
+
```
Comment (fchollet) [inline examples/vision/md/semisupervised_simclr.md:1]: There's an unnecessary blank line here. Beyond that, there seems to be a lot of lines that are marked as modified as git but that shouldn't be; any clue as to why? Some whitespace changes?

Comment (johko) [reply examples/vision/md/semisupervised_simclr.md:1]: You're right. It seems a bit strange, but in general I did not change a big deal in formatting. When using a local difftool for comparison it also does not show as many changes as here in GitHub, maybe that is some markdown GitHub issue.

I created the files with the suggested `python autogen.py add_example vision/semisupervised_simclr.py`

**Review by fchollet — APPROVED** [review]
LGTM, thanks!
